cli/loader_lib.c

```diff
@@ -55,19 +67,21 @@ static void * load_library(const char * rel_path, const char * src_dir, int err)
     strncat(path, rel_path, sizeof(path) - 1);
 
 #if defined(_OS_WINDOWS_)
+#define PATH_EXISTS() win_file_exists(wpath)
     wchar_t wpath[2*JL_PATH_MAX + 1] = {0};
     if (!utf8_to_wchar(path, wpath, 2*JL_PATH_MAX)) {
         jl_loader_print_stderr3("ERROR: Unable to convert path ", path, " to wide string!\n");
         exit(1);
     }
     handle = (void *)LoadLibraryExW(wpath, NULL, LOAD_WITH_ALTERED_SEARCH_PATH);
 #else
+#define PATH_EXISTS() !access(path, F_OK)
     handle = dlopen(path, RTLD_NOW | (err ? RTLD_GLOBAL : RTLD_LOCAL));
 #endif
-
     if (handle == NULL) {
-        if (!err)
+        if (!err && !PATH_EXISTS())
             return NULL;
+#undef PATH_EXISTS
         jl_loader_print_stderr3("ERROR: Unable to load dependent library ", path, "\n");
```

Comment:
Should use `wpath` on Windows?

Reply:
It should be utf8. The function expects it to be utf8. That it isn't is very odd, and this PR made no changes that would break that.

Reply:
Yes, but the tests introduced here triggered the error, [tests were already failing in this PR](https://buildkite.com/julialang/julia-master/builds/17576#01841ed4-f497-4e8b-98c1-f2a6fec2e9d5/592-1457), but the fact that Windows tests are allowed to fail means that the failure was overlooked